release-note-script/src/main/java/ReleaseNoteCreation.java

```diff
@@ -0,0 +1,331 @@
+      String topicPrNumber = entry.getKey();
+      List<ReleaseNoteText> releaseNoteTextsInSameAs = entry.getValue();
+
+      Arrays.stream(Category.values())
+          .forEach(
+              category -> {
+                List<ReleaseNoteText> releaseNoteTextsInACategory = categoryMap.get(category);
+                if (releaseNoteTextsInACategory != null) {
+                  releaseNoteTextsInACategory.forEach(
+                      rnInfo -> {
+                        if (rnInfo.prNumbers.get(0).equals(topicPrNumber)) {
+                          releaseNoteTextsInSameAs.forEach(from -> merge(from, rnInfo));
+                        }
+                      });
+                }
+              });
+    }
+  }
+
+  private void merge(ReleaseNoteText from, ReleaseNoteText to) {
+    if (from.text != null && !from.text.isEmpty()) {
+      to.text = to.text + " " + from.text;
+    }
+    if (DEBUG != null) System.err.printf("merged RN text: %s%n", to.text);
+    to.prNumbers.addAll(from.prNumbers);
+  }
+
+  void outputReleaseNote() {
+    StringBuilder builder = new StringBuilder();
+    builder.append("## Summary\n\n");
+
+    Arrays.stream(Category.values())
+        .forEach(
+            category -> {
+              List<ReleaseNoteText> releaseNotes = categoryMap.get(category);
+              if (releaseNotes != null && !releaseNotes.isEmpty()) {
+                builder.append(String.format("## %s%n", category.getDisplayName()));
+                builder.append(getFormattedReleaseNotes(releaseNotes)).append("\n");
+              }
+            });
+    System.out.println(builder);
+  }
+
+  private String getFormattedReleaseNotes(List<ReleaseNoteText> releaseNotes) {
+    StringBuilder builder = new StringBuilder();
+    releaseNotes.forEach(
+        rnText -> {
+          builder.append(String.format("- %s (", rnText.text));
+          rnText.prNumbers.forEach(prNum -> builder.append(String.format("#%s ", prNum)));
+          builder.deleteCharAt(builder.length() - 1); // delete the last space character
+          builder.append(")\n");
+        });
+    return builder.toString();
+  }
+
+  private void checkReleaseNoteCategory(ReleaseNoteText rnText) {
+    if (rnText.category == null) {
+      rnText.category = Category.MISCELLANEOUS;
+    }
+  }
+
+  private boolean isContainedInSameAsItems(ReleaseNoteText rnText) {
+    return sameAsItems.values().stream().anyMatch(items -> items.contains(rnText));
+  }
+
+  public enum Category {
+    ENHANCEMENT("Enhancements"),
+    IMPROVEMENT("Improvements"),
+    BUGFIX("Bug fixes"),
+    DOCUMENTATION("Documentation"),
+    MISCELLANEOUS("Miscellaneous");
+
+    private final String displayName;
+
+    Category(String displayName) {
+      this.displayName = displayName;
+    }
+
+    public String getDisplayName() {
+      return this.displayName;
+    }
+
+    public static Category getByName(String name) {
```

Comment:
This is an aesthetic point. Since `get` prefix is used to define a getter to access private fields, it would be better to use this kind of factory method, like `Category.fromName`.

Reply:
I see. I'll revise the method name. Thank you

Reply:
Done. 8fe1f0f